docs/xks/developer-guide/gitops.md

```diff
@@ -0,0 +1,71 @@
+  - env: prod
+    auto: false
+```
+
+The `dev` and `qa` environemnts have `auto: false` which means that new releases will be automatically applied, while `prod` environment is configured with `auto: false`. This means that pull requests must be merged by a human, presumably one that has verified that the update worked as expected in previous environments. The GitOps repository is configured to require checks on pull requests to pass in order to allow merge.
+
+### Applying to dev
+
+The flow is fully automatic and is triggered by the container image upload.
+
+<img alt="Apply to dev" src={useBaseUrl("img/assets/xks/developer-guide/developer-flow-apply-dev.jpg")} />
+
+1. The container image upload triggers a pipeline in the GitOps repository that runs the [gitops-promotion new](https://github.com/XenitAB/gitops-promotion#gitops-promotion-new) command. It pushes a new branch and updates the `dev` environment manifest for the app with the new tag. It then opens an "auto-merging" pull request to integrate the new tag into the main branch.
+1. The pull request triggers another pipeline that runs [gitops-promotion status](https://github.com/XenitAB/gitops-promotion#gitops-promotion-new) command. Since `dev` is the first environment in the list, it does nothing and reports success.
+1. The pull request check turns green and the pull request is automatically merged by Git provider.
+1. The Flux Kustomization controller detects that there has been an update to the app's tag in the Git repository and applies this update to the Kubernetes resource for the app (typically a `Deployment`).
+1. The pods running the new container image came up healthy and so Flux sets a commit status on the `main` branch in the GitOps repository, reporting that the update was successfully applied. This will be significant in the next section.
+
+### Applying to qa
+
+<img alt="Apply to qa" src={useBaseUrl("img/assets/xks/developer-guide/developer-flow-apply-qa.jpg")} />
+
+1. Merging a promotion to the main branch triggers a pipeline in the GitOps repository that runs the [gitops-promotion promote](https://github.com/XenitAB/gitops-promotion#gitops-promotion-promote) command. Like `new`, it creates a branch and updates the `qa` envrionment manifest for the app with the new tag. Because the configuration for this environment says `auto: true` it creates an auto-merging pull request.
+1. As before, this new pull request triggers antoher pipeline that runs the `status` command. This time there is a previous environment and the status command reads the Flux commit status for that environment. Since Flux managed to apply the change in `dev` the `status` command reports success.
+1. The pull request check turns green and the pull request is automatically merged by Git provider.
+1. The Flux Kustomization controller detects that there has been an update to the app's tag in the Git repository and applies this update to the Kubernetes resource for the app (typically a Deployment).
```

Comment:
I think we should put Deployment within a code snippet like line 44 (`Deployment` instead of Deployment)

Reply:
Fixed.